File: src/CENTRO_SALUD/Atencion.java
package CENTRO_SALUD;

import java.util.ArrayList;
import java.util.List;

public class Atencion {
    private String fecha;
    private String motivo;
    private String diagnostico;
    private String tratamiento;
    private Profesional profesional; // quien atiende
    private HistoriaClinica historiaClinica; // a que historia pertenece
    private FUA fua;   // documento generado 
    private List<RecetaMedica> recetas = new ArrayList<>(); // 1 atencion -> N recetas
 
    public Atencion(String fecha, String motivo, String diagnostico, String tratamiento,
                     Profesional profesional, HistoriaClinica historiaClinica) {
        this.fecha = fecha;
        this.motivo = motivo;
        this.diagnostico = diagnostico;
        this.tratamiento = tratamiento;
        this.profesional = profesional;
        this.historiaClinica = historiaClinica;
    }

    public void agregarReceta(RecetaMedica receta) {
        recetas.add(receta);
    }
 
    public String getFecha() { return fecha; }
    public String getMotivo() { return motivo; }
    public String getDiagnostico() { return diagnostico; }
    public String getTratamiento() { return tratamiento; }
    public Profesional getProfesional() { return profesional; }
    public HistoriaClinica getHistoriaClinica() { return historiaClinica; }
 
    public FUA getFua() { return fua; }
    public void setFua(FUA fua) { this.fua = fua; }

    public List<RecetaMedica> getRecetas() { return recetas; }
 
}

File: centro_salud_python/atencion.py
class Atencion:
    """Una atencion medica: motivo, diagnostico, tratamiento y quien la registro."""

    def __init__(self, fecha, motivo, diagnostico, tratamiento, profesional, historia_clinica):
        self._fecha = fecha
        self._motivo = motivo
        self._diagnostico = diagnostico
        self._tratamiento = tratamiento
        self._profesional = profesional            # quien atiende
        self._historia_clinica = historia_clinica  # a que historia pertenece
        self._fua = None                            # documento generado
        self._recetas = []                          # 1 atencion -> N recetas

    def agregar_receta(self, receta):
        self._recetas.append(receta)

    def get_fecha(self):
        return self._fecha

    def get_motivo(self):
        return self._motivo

    def get_diagnostico(self):
        return self._diagnostico

    def get_tratamiento(self):
        return self._tratamiento

    def get_profesional(self):
        return self._profesional

    def get_historia_clinica(self):
        return self._historia_clinica

    def get_fua(self):
        return self._fua

    def set_fua(self, fua):
        self._fua = fua

    def get_recetas(self):
        return self._recetas
